### PR TITLE
Add support for *_HOME variables in rustup.

### LINKED ISSUE #5013: [Bug]:
### Prerequisites

- [X] I have written a descriptive issue title.
- [X] I have searched all issues/PRs to ensure it has not already been reported or fixed.
- [X] I have verified that I am using the latest version of Scoop and corresponding bucket.

### Package Name

rustup

### Expected/Current Behaviour

With the introduction of [Windows Dev Drive](https://learn.microsoft.com/en-us/windows/dev-drive/), I want to move cargo folder as per Microsoft recommendation.
Scoop install always overrides CARGO_HOME, so I can't change folder to custom location.

### Steps to Reproduce

```console
setx CARGO_GOME E:/packages/cargo
scoop install rustup
echo $env:CARGO_HOME # This results in C:\Users\user\scoop\persist\rustup\.cargo
```

### Possible Solution

Do not overwrite environment variable
https://github.com/ScoopInstaller/Main/blob/727f1212dc401ee658f1a313ccbd7974370673f9/bucket/rustup.json#L33-L35
Pass existing environment variables
https://github.com/ScoopInstaller/Main/blob/727f1212dc401ee658f1a313ccbd7974370673f9/bucket/rustup.json#L26C12-L29

### Scoop and Buckets Version

```console
Current Scoop version:
v0.3.1 - Released at 2022-11-15

'extras' bucket:
989df8fb9 (HEAD -> master, origin/master, origin/HEAD) steamcmd: Update to version 1691628584

'java' bucket:
2e9a6179 (HEAD -> master, origin/master, origin/HEAD) temurin11-nightly-jre: Update to version 11.0.21-1.0.202308081844

'main' bucket:
727f1212d (HEAD -> master, origin/master, origin/HEAD) red: Update to version 10aug23

'versions' bucket:
6db6c0f9a (HEAD -> master, origin/master, origin/HEAD) vlc-nightly: Update to version 20230810
```

### Scoop Config

```console
scoop config

last_update         scoop_branch scoop_repo
-----------         ------------ ----------
10/08/2023 10:17:13 master       https://github.com/ScoopInstaller/Scoop
```

### PowerShell Version

```console
$PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.3.6
PSEdition                      Core
GitCommitId                    7.3.6
OS                             Microsoft Windows 10.0.22631
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Additional Softwares

_No response_

### PR DESCRIPTION
**Pull request by marknefedov**
Only set CARGO_HOME and RUSTUP_HOME environment variables if not set already.
If vars are set, install to those location.

Closes #5013 

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by rashil2000** [comment]
Hi @marknefedov!

Why did you close?

**Comment by marknefedov** [comment]
In the issue it have been suggested to move to a separate manifest, and
apparently the last command in the post install script is unreliable, or it
is my windows installation being finicky. Feel free to use as a base though.

File: bucket/rustup.json [hunk 1]
```diff
@@ -24,16 +24,24 @@
...
-        "RUSTUP_HOME": "$persist_dir\\.rustup"
-    },
+    "post_install" :[
+            "if ($null -eq [Environment]::GetEnvironmentVariable('CARGO_HOME', 'Process')) {",
+            "   setx CARGO_HOME \"$persist_dir\\.cargo\"",
+            "}",
+            "if ($null -eq [Environment]::GetEnvironmentVariable('RUSTUP_HOME', 'Process')) {",
+            "   setx RUSTUP_HOME \"$persist_dir\\.rustup\"",
+            "}",
+            "setx PATH \"%PATH%;%CARGO_HOME%\\bin\""
```
Comment (brian6932) [inline bucket/rustup.json:43]:
```suggestion
            "setx PATH \"$env:PATH;$env:CARGO_HOME\\bin\""
```
This is PowerShell not cmd, prob why it seemed "unreliable", wrong syntax

**Review by brian6932 — CHANGES_REQUESTED** [review]

**Comment by HUMORCE** [comment]
I keep my opinion that Scoop should manage the environment variables of Apps as possible. If other maintainers would like to merge it, feel free to go.

**Pull request closed without merging**